packages/react-composites/scripts/runBrowserTests.mjs

```diff
@@ -128,13 +150,48 @@ async function runOne(args, composite, hermeticity) {
   if (args.dryRun) {
     console.log(`DRYRUN: Would have run ${cmd}`);
   } else {
-    await exec(cmd, env);
+    await exec(cmd, extraEnv);
   }
 }
 
 function setup() {
   console.log('Cleaning up output directory...');
-  rmdirSync(BASE_OUTPUT_DIR, { recursive: true });
+  fs.rmSync(BASE_OUTPUT_DIR, { recursive: true, force: true });
+}
+
+async function preprocessTests() {
+  fs.rmSync(PREPROCESSED_ROOT, { recursive: true, force: true });
+  await createTsconfigTrampoline();
+  await preprocessDir(SRC_ROOT, PREPROCESSED_SRC_ROOT);
+  await preprocessDir(path.join(TEST_ROOT, 'tests'), path.join(PREPROCESSED_TEST_ROOT, 'tests'));
+}
+
+async function createTsconfigTrampoline() {
+  fs.mkdirsSync(PREPROCESSED_ROOT);
+  fs.writeFileSync(path.join(PREPROCESSED_ROOT, 'tsconfig.json'), '{ "extends": "../tsconfig.preprocess.json" }');
+}
+
+async function preprocessDir(fromDir, toDir) {
+  console.log(`Preprocessing ${fromDir} to ${toDir}`);
+  fs.mkdirsSync(toDir);
+  fs.copySync(fromDir, toDir, { overwrite: true, preserveTimestamps: true });
+  await exec(
+    quote([
+      'npx',
```

Comment:
Maybe we can execute command here from package.json file like `npm run preprocess: test` or so (with 2nd parameter)? Good to have a centralized place to store all the commands (I can feel some other reasons we put command here as well, so highly depends on you)

Reply:
The intention is to keep moving logic from `package.json` `scripts` object to actual node scripts.

We can extract common logic as we use it.